### ISSUE TITLE
Assembly tracks in IGV

### ISSUE THREAD

**Issue by fellen31** [Clinical-Genomics/scout#4982]
Hi,

What do you think about the possibility to view the mapped assembly as tracks in IGV? That could be shown (next to the reads), if files are included in the scout config? 

Currently Nallo outputs each haplotype into separate bam-files.

**Comment by dnil** [comment]
Hi, could you link a corresponding PR/issue on Nallo or the tool you use to produce these to make it easier to follow?

It sounds like you do a de novo assembly into two separate bam files. Are these with their own consensus, or already aligned back to a reference genome? If the latter, this should be relatively straightforward, though it is a little unclear why we don't just use haplotype tagging in the joint cram instead? That was what we had previously, right?

**Comment by fellen31** [comment]
> If the latter, this should be relatively straightforward, though it is a little unclear why we don't just use haplotype tagging in the joint cram instead?

Yes, within Nallo hifiasm produces a dual de novo assembly output as two fastq-files, these are then aligned back to the reference and saved as bam files. If it suits scout better, I could look into adding tags to the contigs and joining the bam files. 

> That was what we had previously, right?

I don't think so :)

**Comment by fellen31** [comment]
> Hi, could you link a corresponding PR/issue on Nallo or the tool you use to produce these to make it easier to follow?

I created an [issue](https://github.com/genomic-medicine-sweden/nallo/issues/442), if you think it's better discussed there.

**Comment by dnil** [comment]
> > That was what we had previously, right?
> 
> I don't think so :)

We have looked at WhatsHap haplotype tags in another issue, on a test case loaded from Nallo, so was under the impression that was the state of the art a little bit back?

**Comment by fellen31** [comment]
For aligned reads, yes, reads come in as a single file, are tagged with the help of phased variants and output as a single file. But for the aligned assemblies, as they are output as two separate files from the beginning, there has not yet been a need to tag them.

**Comment by dnil** [comment]
Right, got that part, I mean it is very similar data? We had the HP info from WhatsHap before.

Like so: additional tracks can certainly be arranged - you will need to register output types with Hermes, pass them to Housekeeper, export them through CG. Given the key names we can add them to the load model, and to track generation. All good. Tags would be immediately available.

But the real question is perhaps in the quality of the data? If de novo data remapped to reference is better overall than the reference guided/mapped ones perhaps we should just replace the cram in the first place?

**Comment by dnil** [comment]
Oh, does the `hifiasm` output include the underlying reads as well, or is it "only" the haplotype consensus?

**Comment by fellen31** [comment]
> Right, got that part, I mean it is very similar data? We had the HP info from WhatsHap before.

Yes!

> Like so: additional tracks can certainly be arranged - you will need to register output types with Hermes, pass them to Housekeeper, export them through CG. Given the key names we can add them to the load model, and to track generation. All good. Tags would be immediately available.

Sounds good! I'll hack together a test-case with HP-tags to upload instead of the normal bam, so we can see what it looks like. 

> But the real question is perhaps in the quality of the data? If de novo data remapped to reference is better overall than the reference guided/mapped ones perhaps we should just replace the cram in the first place?

I don't think we can do that switch (yet). I think in some regions the assembly will be better, in others the reads will be more informative. But I believe they are complementary, and might help/serve as a control if visually inspecting variants. 

> Oh, does the `hifiasm` output include the underlying reads as well, or is it "only" the haplotype consensus?

No, it's only the consensus sequence.

**Comment by dnil** [comment]
Sounds good! My thoughts would be that it is not super useful to have only the concensus in isolation from the assembly graph, reads or some sort of quality scores. Some regions will of course be super solid, whereas others are separated by regions of a bit more tentative scaffolding, and consensus calling likewise will be distinct for the more unique parts of the genome, then a bit more fuzzy for the more difficult parts. As good a combo as we can get without too much effort is good: of course it helps to have the de novo concensus tracks as one more source of evidence when eg considering if two variants are on the same allele or not.

**Comment by fellen31** [comment]
Uploading in one file with HP-tags seems to works well, it would be a fairly minor addition to nallo. Let me know what you if you think this is what you would prefer. 

<img width="1421" alt="Screenshot 2024-10-28 at 14 42 46" src="https://github.com/user-attachments/assets/179f8ccb-0893-46c6-8323-54844ced3bab">

**Comment by fellen31** [comment]
This PR would add a mapped and HP-tagged BAM-file to Nallo: https://github.com/genomic-medicine-sweden/nallo/pull/566

> Like so: additional tracks can certainly be arranged - you will need to register output types with Hermes, pass them to Housekeeper, export them through CG. Given the key names we can add them to the load model, and to track generation. All good. Tags would be immediately available.

Do you need key names before updating the scout load model and track generation, or could you update scout first?

**Comment by dnil** [comment]
No need, we can come up with something, but suggestions are always welcome. E.g. `assembly_alignment_path`?

**Comment by fellen31** [comment]
> `assembly_alignment_path`

If @peterpru doesn't have any objections, `assembly_alignment_path` sounds very reasonable to me!

**Comment by peterpru** [comment]
sounds good!
